azure/services/subnets/subnets_test.go

```diff
@@ -338,6 +362,122 @@ func TestReconcileSubnets(t *testing.T) {
 				}).Times(1)
 			},
 		},
+		{
+			name:          "doesn't overwrite existing NAT Gateway",
```

Comment:
@CecileRobertMichon is this a correct UT to express the existing NAT Gateway expectation?

Reply:
looks good